core/Pos/Binary/Crypto.hs

```diff
@@ -87,12 +88,23 @@ instance Bi SecretSharingExtra where
           <*> decode
           <*> decode
 
-deriving instance Bi (AsBinary SecretSharingExtra)
-
 ----------------------------------------------------------------------------
 -- SecretSharing AsBinary
 ----------------------------------------------------------------------------
 
+-- !A note about these instances! --
+--
+-- For most of the secret sharing types the only check we do during
+-- deserialization is length check. As long as length matches our
+-- expectations, the decoding succeeds (look at 'Binary' instances in
+-- 'pvss') which in turn means that we can use 'fromBinary' and be
+-- quite sure it will succeed. That's why it's important to check
+-- length here (this check is cheap, so it's good to do it as soon as
+-- possible). The only exception is 'SecretSharingExtra' (because we
+-- don't know its length in advance). Currently we check that
+-- 'SecretSharingExtra' can be parsed in 'verifyCommitment' function.
+--
+-- FIXME [CSL-1412] Check length here (the note above is currently incorrect)!
```

Comment:
@gromakovsky Just as an heads-up, I'm planning to tackle https://issues.serokell.io/issue/CSL-1412 today. It would be awesome if you could chime in to that ticket and review the solution I'm proposing. Up to you then if you want to include this comment in this PR (I don't have a strong opinion).